framework/tests/flowexpression/InnerClasses.java

```diff
@@ -0,0 +1,38 @@
+package flowexpression;
+
+import testlib.flowexpression.qual.FlowExp;
+
+public class InnerClasses {
+    public String outerInstanceField = "";
+    public static String outerStaticField = "";
+
+    static class InnerClass {
+        // :: error: (expression.unparsable.type.invalid)
+        @FlowExp("outerInstanceField") Object o = null;
+
+        @FlowExp("outerStaticField") Object o2 = null;
+    }
+
+    class NonStaticInnerClass {
+        @FlowExp("outerInstanceField") Object o = null;
+
+        @FlowExp("outerStaticField") Object o2 = null;
+    }
+
+    static class InnerClass2 {
+        public String outerInstanceField = "";
+
+        @FlowExp("outerInstanceField") Object o = null;
+    }
+
+    class TestUses {
+        void method(InnerClass innerClass, InnerClass2 innerClass2) {
+            // :: error: (expression.unparsable.type.invalid) :: error:
+            // (assignment.type.incompatible)
```

Comment:
@smillst I'm not sure why an `expression.unparsable.type.invalid` error is expected here, since the inner class `TestUses` and the method `method` are non-static. Can you please explain why you think it should be issued?
I agree with the `assignment.type.incompatible` error, but I think this is issue #152, `framework/tests/flowexpression/ThisSuper.java` test case. In the last part of the test case, something similar is tested and it fails.

Reply:
The error should be on the explicitly written annotation `@FlowExp("innerClass.outerInstanceField")`.  
The type of `innerClass` is `InnerClass`. `outerInstanceField` can't be referenced from an instance of `InnerClass`.  Also, if I had written innerClass.outerInstanceField, javac would issue an error.